src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFiltersCategory.tsx

```diff
@@ -25,9 +25,13 @@ const GlobalLearningResourcesFiltersCategory: React.FC<FiltersCategory> = ({
   };
 
   const updateLoaderOptions = (filter: Filter, isChecked: boolean) => {
+    const currentCategory = loaderOptions[categoryId];
+
     const updatedCategory = isChecked
-      ? [...(loaderOptions[categoryId] || []), filter.id]
-      : (loaderOptions[categoryId] || []).filter((id) => id !== filter.id);
+      ? [...(Array.isArray(currentCategory) ? currentCategory : []), filter.id]
```

Comment:
This is rough to read. I should work but can we make it nicer on the eyes? Don't be afraid of using god ol if statement. Nesting ternary operators can get confusing pretty quickly.

I would not be against just crating a simple function.

Reply:
I created a simple function involving the if else statements.